src/main.ts

```diff
@@ -1,22 +1,21 @@
-import Fastify from 'fastify'
+import Fastify, { FastifyBaseLogger } from 'fastify'
 
 import { createPrismaClient } from './modules/prisma.js'
 import { createRoutes } from './routes/index.js'
 import { createLogger } from './modules/logger.js'
 import { spawnJobs } from './jobs/index.js'
-import { config } from './config.js'
+import { config } from './config/index.js'
 import { createAdamantClient } from './modules/adamantClient.js'
 import { createFcmClient } from './modules/fcmClient.js'
 import { FcmNotification } from './adapters/notification/fcmNotification.js'
-import { BaseNotification } from './adapters/notification/baseNotification.js'
 
 export const main = async () => {
   const { logger } = createLogger()
-  const fastify = Fastify({ logger })
+  const fastify = Fastify({ logger: logger as FastifyBaseLogger })
   const prisma = createPrismaClient()
   const adamantClient = createAdamantClient()
 
-  let notificationService = new BaseNotification()
+  let notificationService
 
   if (config.app.notificationService === 'FCM') {
     const fcmClient = createFcmClient()
```

Comment:
`fcmClient` must be initialized inside `FcmNotification` class

Reply:
Remove `notificationService` from the config. Consider that in the future we will add Apple Push Notifications for iOS as well, so you don't need to bind to specific push notification service. To easy the process of choosing the right instance you can define services as follow:

```ts
type ServiceType = 'FCM' | 'APNS' // you can reuse `PushServiceProvider` enum from Prisma

const pushService: Record<ServiceType, BaseNotificationInterface> = {
  FCM: new FcmNotification(),
  APNS: new ApnsNotification()
}
```

Usage:

```ts
import { pushService } from './push.service' 

pushService['FCM'].message(payload)
pushService['APNS'].message(payload)

const service = pushService[device.serviceProvider]
service.message(payload)
```